packages/blocks/google-cloud/test/get-recommendations-cli-action.test.ts

```diff
@@ -0,0 +1,209 @@
+    });
+  });
+
+  test('should return empty property when unauthenticated', async () => {
+    const context = createContext({});
+
+    const result = await getRecommendationsAction.props[
+      'filterByProperty'
+    ].props(
+      {
+        auth: undefined as any,
+        useHostSession: { useHostSessionCheckbox: false },
+      },
+      context,
+    );
+
+    expect(result['markdown']).toMatchObject({
+      type: 'MARKDOWN',
+      description: 'Please authenticate to see the filter options.',
+    });
+  });
+
+  test('should return markdown property if no filterBySelection is selected', async () => {
+    const context = createContext({});
+
+    const result = await getRecommendationsAction.props[
+      'filterByProperty'
+    ].props(
+      {
+        auth,
+        useHostSession: { useHostSessionCheckbox: true },
+        filterBySelection: undefined as any,
+      },
+      context,
+    );
+
+    expect(result).toMatchObject({
+      markdown: {
+        type: 'MARKDOWN',
+        description:
+          'Please select a filter from the dropdown to see the options.',
+      },
+    });
+  });
+
+  test('should return dynamic dropdown property for billingAccount', async () => {
+    const context = createContext({});
+    gcloudCliMock.runCommand.mockResolvedValue(
+      JSON.stringify([
+        { name: 'billingAccounts/abc', displayName: 'Billing 1' },
+      ]),
+    );
+
+    const result = await getRecommendationsAction.props[
+      'filterByProperty'
+    ].props(
+      {
+        auth,
+        useHostSession: { useHostSessionCheckbox: true },
+        filterBySelection: 'billingAccount' as any,
+      },
+      context,
+    );
+
+    expect(result['billingAccount']).toMatchObject({
+      displayName: 'Billing Account',
+      type: 'STATIC_DROPDOWN',
+    });
+
+    expect(gcloudCliMock.runCommand).toHaveBeenCalledWith(
+      'gcloud billing accounts list --format=json',
+      auth,
+      true,
+    );
+  });
+
+  test('should return dynamic short text property for folder', async () => {
+    const context = createContext({});
+
+    const result = await getRecommendationsAction.props[
+      'filterByProperty'
+    ].props(
+      {
+        auth,
+        useHostSession: { useHostSessionCheckbox: true },
+        filterBySelection: 'folder' as any,
+      },
+      context,
+    );
+
+    expect(result['folder']).toMatchObject({
+      displayName: 'Folder ID',
+      type: 'SHORT_TEXT',
+    });
+
+    expect(gcloudCliMock.runCommand).not.toHaveBeenCalled();
+  });
+
+  test('should build correct command and parse responses for multiple recommenders', async () => {
+    const context = createContext({
+      useHostSession: { useHostSessionCheckbox: true },
+      filterByProperty: { organization: 'org-789' },
+      location: 'us-east1',
+      recommenders: ['recommender-1', 'recommender-2'],
+    });
+
+    gcloudCliMock.runCommands.mockResolvedValueOnce([
```

Comment:
try also a recommender that returns nothing, and a recommender that returns 2 instead if 1. To make sure that the flattening works properly.